peak/mapper/binding.py

```diff
@@ -82,7 +82,7 @@ def __init__(self,
         arch_isa : Product,
         ir_isa : Product,
         allow_existential : bool, #allow unbound to be Existential
-        custom_enumeration : tp.Mapping[type, tp.Callable] = {}
+        custom_enumeration : tp.Mapping[type, tp.Callable] = None
     ):
         self.enumeration_scheme = SubTypeDict(custom_enumeration)
```

Comment:
Need 
```
if custom_enumeration is None:
  custom_enumeration = {}
```

Reply:
Why? You have that logic in the __init__ of SubTypeDict

Reply:
Also the fact that no test caught this bug is concerning

Reply:
Still not sure what bug you are referring to. the init of SubTypeDict contains the logic if custom_enumeration is None.

Reply:
Ahh that is true, that is bug SubTypeDict.  It should not default to None, it should default to ().  I will correct it.

Reply:
The bug I was referring to was what happens when custom_enumeration is None. I assumed I had made SubTypeDict behave like dict where dict(None) is an error.

Reply:
Looks like you fixed this